perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/MembersManagerBlImpl.java

```diff
@@ -297,6 +300,36 @@ public Member createSpecificMember(PerunSession sess, Vo vo, Candidate candidate
 		return member;
 	}
 
+	public Member createSponsoredAccount(PerunSession sess, Map<String, String> params, String namespace, ExtSource extSource, String extSourcePostfix, User owner, Vo vo) throws InternalErrorException, ExtendMembershipException, AlreadyMemberException, WrongReferenceAttributeValueException, WrongAttributeValueException, GroupOperationsException, PasswordCreationFailedException, ExtSourceNotExistsException, LoginNotExistsException, UserNotExistsException {
+		String loginNamespaceUri = "urn:perun:user:attribute-def:def:login-namespace:" + namespace;
+		boolean passwordPresent = params.get("password") != null;
+		if(params.get(loginNamespaceUri) == null) {
+			Map<String, String> generatedParams = getPerunBl().getUsersManagerBl().generateAccount(sess, namespace, params);
+			params.putAll(generatedParams);
+		} else if (passwordPresent) {
+			getPerunBl().getUsersManagerBl().reservePassword(sess, params.get(loginNamespaceUri), namespace, params.get("password"));
+		} else {
+			throw new WrongAttributeValueException("If login for new account is provided, password must be provided also");
```

Comment:
WrongAttributeValueException is used exclusively for problems with perun attributes. Please use different exception.